bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/ViewsPreferencePage.java

```diff
@@ -373,13 +410,20 @@ public boolean performOk() {
 		if (themingEnabledChanged) {
 			showRestartDialog();
 		}
+		if (isUpdateOnRuntimeChanged) {
+			showRestartDialog(WorkbenchMessages.RuntimeSettingChangeWarningTitle,
+					WorkbenchMessages.RuntimeSettingChangeWarningDesc);
+		}
 
 		return super.performOk();
 	}
 
 	private void showRestartDialog() {
-		if (new MessageDialog(null, WorkbenchMessages.ThemeChangeWarningTitle, null,
-				WorkbenchMessages.ThemeChangeWarningText, MessageDialog.NONE, 2,
+		showRestartDialog(WorkbenchMessages.ThemeChangeWarningTitle, WorkbenchMessages.ThemeChangeWarningText);
+	}
+
+	private void showRestartDialog(String title, String warningText) {
+		if (new MessageDialog(null, title, null, warningText, MessageDialog.NONE, 2,
```

Comment:
Please properly refactor this to either always have the consumer decise which title and text to put in the dialog or to have one parameterless method per use case. Now we have one method for custom title/text and one with some unspecified default values (which by accident default to themeing). I would propose to just have the method with two parameters and make the method consumers pass the required information.